pkg/systemtemplate/import.go

```diff
@@ -136,8 +135,8 @@ func PodDisruptionBudgetTemplate(cluster *apimgmtv3.Cluster) ([]byte, error) {
 func SystemTemplate(resp io.Writer, agentImage, authImage, namespace, token, url string, isWindowsCluster, isPreBootstrap bool, cluster *apimgmtv3.Cluster,
 	agentFeatures map[string]bool, taints []corev1.Taint, secretLister v1.SecretLister, pcExists bool) error {
 	var tolerations, agentEnvVars, agentAppendTolerations, agentAffinity, agentResourceRequirements string
-	d := md5.Sum([]byte(url + token + namespace))
-	tokenKey := hex.EncodeToString(d[:])[:7]
+	d := sha256.Sum256([]byte(fmt.Sprintf("%s.%s.%s", url, token, namespace)))
+	tokenKey := hex.EncodeToString(d[:])[:10]
```

Comment:
What happens to the existing `cattle-credentials-` secrets?

Reply:
hmmm, I feel like we should prevent multiple secrets from being left around. The more unknown resources we have, the harder it will be to clean up.